proposals/csharp-9.0/nullable-reference-types-specification.md

```diff
@@ -49,29 +69,75 @@ primary_constraint
     ;
 ```
 
+A type parameter constrained with `class` (in an *enabled* annotation context) must be a nonnullable reference type.
+
+A type parameter constrained with `class?` (or `class` in a *disabled* annotation context) may either be a nullable or nonnullable reference type.
+
+A warning is given on a `class?` constraint in a *disabled* annotation context.
```

Comment:
May be worth calling out explicitly that `class` is okay here but `class?` is not. Think it's clearer that way.